.github/workflows/build.yml

```diff
@@ -114,3 +114,37 @@ jobs:
       - run: ./tool/shared/deploy.sh --robots ok default
         env:
           FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
+
+  deploy_preview:
+    needs: test
+    runs-on: ubuntu-latest
+    if: '${{ github.event.pull_request.head.repo.full_name == github.repository }}'
```

Comment:
Does this limit the previews to PRs from branches created on this repository?

Reply:
Yes, that is my understanding

Reply:
To illustrate this a bit, the source for this PR is mit-mit:firebasepreview so the steps here weren't actually run. So I would suggest to re-create the PR from a branch in this repo so the effects can be seen before landing this (but read the other comment before).

Reply:
I think that would mean adding if's to each and every step here, as we don't want to run the build, etc. unless we're doing a push to master or a PR from a non-fork?